projects/js-packages/ai-client/src/components/ai-control/block-ai-control.tsx

```diff
@@ -0,0 +1,281 @@
+/**
+ * External dependencies
+ */
+import { Button, ButtonGroup } from '@wordpress/components';
+import { useKeyboardShortcut } from '@wordpress/compose';
+import { useImperativeHandle, useRef, useEffect, useCallback, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import {
+	Icon,
+	closeSmall,
+	check,
+	arrowUp,
+	trash,
+	reusableBlock as regenerate,
+} from '@wordpress/icons';
+import debugFactory from 'debug';
+import React, { forwardRef } from 'react';
+/**
+ * Internal dependencies
+ */
+import { GuidelineMessage } from '../message/index.js';
+import AIControl from './ai-control.js';
+import './style.scss';
+/**
+ * Types
+ */
+import type { RequestingStateProp } from '../../types.js';
+import type { ReactElement } from 'react';
+
+type BlockAIControlProps = {
+	disabled?: boolean;
+	value: string;
+	placeholder?: string;
+	showAccept?: boolean;
+	acceptLabel?: string;
+	showButtonLabels?: boolean;
+	isTransparent?: boolean;
+	state?: RequestingStateProp;
+	showGuideLine?: boolean;
+	customFooter?: ReactElement;
+	onChange?: ( newValue: string ) => void;
+	onSend?: ( currentValue: string ) => void;
+	onStop?: () => void;
+	onAccept?: () => void;
+	onDiscard?: () => void;
+	showRemove?: boolean;
+	bannerComponent?: ReactElement;
+	errorComponent?: ReactElement;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+const debug = debugFactory( 'jetpack-ai-client:ai-control' );
+
+/**
+ * BlockAIControl component. Used by the AI Assistant block, adding logic and components to the base AIControl component.
+ *
+ * @param {BlockAIControlProps} props  - Component props
+ * @param {React.MutableRefObject} ref - Ref to the component
+ * @returns {ReactElement}             Rendered component
+ */
+export function BlockAIControl(
+	{
+		disabled = false,
+		value = '',
+		placeholder = '',
+		showAccept = false,
+		acceptLabel = __( 'Accept', 'jetpack-ai-client' ),
+		showButtonLabels = true,
+		isTransparent = false,
+		state = 'init',
+		showGuideLine = false,
+		customFooter = null,
+		onChange = noop,
+		onSend = noop,
+		onStop = noop,
+		onAccept = noop,
+		onDiscard = null,
```

Comment:
why is this null? Shouldn't it be noop like the other ones? On the props type for onDiscard we have `() => void`

Reply:
Good catch! There is a check for `onDiscard` on line 178, that must be the reason. That said, since all calls to `on*` are prefixed with `?.`, we can remove this `noop` entirely.